cpp/src/arrow/compute/api_vector.h

```diff
@@ -95,6 +95,13 @@ enum class NullPlacement {
   AtEnd,
 };
 
+enum class Tiebreaker {
+  Lowest,
```

Comment:
I have no strong feelings on this but I'll note that different terminology is used elsewhere:
* [Scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rankdata.html) uses "min", "max", "ordinal", "dense" respectively
* [R](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/rank) uses "min", "max", "first" and doesn't seem to have an equivalent for "dense"